src/BMPDecoder.ts

```diff
@@ -21,45 +23,123 @@ export default class BMPDecoder {
     this.pixelDataOffset = this.bufferData.skip(8).readUint32();
     this.width = this.bufferData.skip(4).readUint32();
     this.height = this.bufferData.readUint32();
-    this.bitDepth = this.bufferData.seek(28).readUint16();
+    this.bitsPerPixel = this.bufferData.seek(28).readUint16();
+    if (this.bitsPerPixel === 16 || this.bitsPerPixel === 4) {
+      throw new Error('4 and 16 bits per pixel are not supported.');
+    }
```

Comment:
Would use a white list (check it uses a value we support, rather than a value we don't support). You might be missing other valid values which we don't support.